File: examples/hello.js
/**
 * Git学习项目 - JavaScript示例文件
 * 这个文件用于演示Git的基本操作
 */

/**
 * 问候函数
 * @param {string} name - 要问候的名字，默认为"World"
 * @returns {string} 问候语
 */
function greet(name = "World") {
    return `Hello, ${name}!`;
}

/**
 * 主函数
 */
function main() {
    console.log(greet());
    console.log(greet("Git学习者"));
    console.log("欢迎来到Git学习项目！");
}

// 如果是在Node.js环境中运行
if (typeof require !== 'undefined' && require.main === module) {
    main();
}

// 导出函数供其他模块使用
if (typeof module !== 'undefined' && module.exports) {
    module.exports = { greet, main };
}

File: examples/hello.py
#!/usr/bin/env python3
"""
Git学习项目 - Python示例文件
这个文件用于演示Git的基本操作
"""

def greet(name="World"):
    """
    问候函数
    
    Args:
        name (str): 要问候的名字，默认为"World"
    
    Returns:
        str: 问候语
    """
    return f"Hello, {name}!"

def main():
    """主函数"""
    print(greet())
    print(greet("Git学习者"))
    print("欢迎来到Git学习项目！")

if __name__ == "__main__":
    main()
